aya/src/maps/mod.rs

```diff
@@ -474,15 +483,15 @@ pub(crate) fn check_v_size<V>(map: &MapData) -> Result<(), MapError> {
 #[derive(Debug)]
 pub struct MapData {
     pub(crate) obj: obj::Map,
-    pub(crate) fd: Option<RawFd>,
+    pub(crate) fd: Option<OwnedFd>,
     pub(crate) btf_fd: Option<RawFd>,
     /// Indicates if this map has been pinned to bpffs
     pub pinned: bool,
 }
 
 impl MapData {
     /// Creates a new map with the provided `name`
-    pub fn create(&mut self, name: &str) -> Result<RawFd, MapError> {
+    pub fn create(&mut self, name: &str) -> Result<(), MapError> {
```

Comment:
Another breaking change. I initially tried to make this return a `BorrowedFd<'_>` to stay similar to the old API but that caused some borrowing issues later since the lifetime of the *mutable* self borrow would be the lifetime of the returned file desriptor which made it awkward to work with. I think this is *okay* since one can ask for the internal file descriptor easy enough.